apps/web-giddh/src/app/new-inventory/component/stock-balance/stock-balance.component.ts

```diff
@@ -1,57 +1,327 @@
+import { Component, OnDestroy, OnInit } from "@angular/core";
 import { FormControl } from "@angular/forms";
+import { select, Store } from "@ngrx/store";
+import { combineLatest, ReplaySubject } from "rxjs";
+import { debounceTime, distinctUntilChanged, takeUntil } from "rxjs/operators";
+import { InventoryService } from "../../../services/inventory.service";
+import { AppState } from "../../../store";
+import { IOption } from "../../../theme/ng-virtual-select/sh-options.interface";
+import { WarehouseActions } from "../../../settings/warehouse/action/warehouse.action";
+import { cloneDeep } from "../../../lodash-optimized";
+import { IGroupsWithStocksHierarchyMinItem } from "../../../models/interfaces/groupsWithStocks.interface";
+import { GroupStockReportRequest, StockReportRequest } from "../../../models/api-models/Inventory";
+import { SettingsFinancialYearActions } from "../../../actions/settings/financial-year/financial-year.action";
+import { GeneralService } from "../../../services/general.service";
+import { ToasterService } from "../../../services/toaster.service";
 
-interface Food {
-  value: string;
-  viewValue: string;
-}
-interface quantity {
-  value: string;
-  viewValue: string;
-}
-interface unit {
-  value: string;
-  viewValue: string;
-}
 @Component({
-    selector: 'stock-balance',
-    templateUrl: './stock-balance.component.html',
-    styleUrls: ['./stock-balance.component.scss'],
+  selector: 'stock-balance',
+  templateUrl: './stock-balance.component.html',
+  styleUrls: ['./stock-balance.component.scss'],
 })
 
-export class StockBalanceComponent implements OnInit{
-    
-    /** Image path variable */
-    public imgPath: string = '';
-    
-    toppings = new FormControl();
-    toppingList: string[] = ['Warehouse 1', 'Warehouse 2', 'Warehouse 3', 'Warehouse 4', 'Warehouse 5', 'Warehouse 6'];
-
-    foods: Food[] = [
-    {value: 'small-0', viewValue: 'Small'},
-    {value: 'medium-1', viewValue: 'Medium'},
-    {value: 'large-2', viewValue: 'Large'},
-    {value: 'extra-large-3', viewValue: 'Extra Large'},
-    {value: 'extra-extra-large-4', viewValue: 'Extra Extra Large'},
-  ];
-
-  quantites: quantity[] = [
-    {value: 'dummy-1', viewValue: 'Dummy 1'},
-    {value: 'dummy-2', viewValue: 'Dummy 2'},
-    {value: 'dummy-3', viewValue: 'Dummy 3'},
-  ];
-
-  units: unit[] = [
-    {value: 'pc', viewValue: 'Pc'},
-    {value: 'kg', viewValue: 'Kg'},
-    {value: 'ltr', viewValue: 'Ltr'},
-    {value: 'nos', viewValue: 'Nos'},
-    {value: 'pcs', viewValue: 'Pcs'},
-  ];
-
-
-    ngOnInit() {
-        /** This will use for image format */
-        this.imgPath = isElectron ? 'assets/images/' : AppUrl + APP_FOLDER + 'assets/images/';
+export class StockBalanceComponent implements OnInit, OnDestroy {
+  /** Image path variable */
+  public imgPath: string = '';
+  /** Observable to unsubscribe all the store listeners to avoid memory leaks */
+  private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
```

Comment:
This file is 2 space formatted, make it 4 space formatted. Make sure you have `indent_size` value set to 4 in `.editorconfig` file. Also check the VS Code default formatting (for more info refer [here](https://www.kindacode.com/article/vs-code-how-to-change-indentation-2-spaces-4-spaces/))

Reply:
@prateek-giddh sir this settings is already applied on VS code settings and .editorconfig file